Services/service_response.c

```diff
@@ -32,59 +32,67 @@ extern Service_Queues_t service_queues;  // Implemented by the host platform
 /**
  * @brief
  * 		Wait on a queue of responses to be sent to other CSP nodes
- * (usually the ground)
+ *              (usually the ground)
  * @details
  * 		CSP client server will wake when data is in the queue for
- * downlink (telemetery)
+ *              downlink (telemetery)
  * @param void * param
  * 		Not used
  * @return
- * 		should not return
+ * 		A SAT_returnState depending on whether we were successful
+ *              in connecting. Not typically used for anything.
  */
-void service_response_task(void *param) {
+SAT_returnState service_response_task(void *param) {
```

Comment:
Since this is a task, and will never return, this should remain void to be consistent with the FreeRTOS task prototypes

Reply:
What about the `return` [on line 66?](https://github.com/hughbagan/ex2_services/blob/response_server/Services/service_response.c#L66)

Reply:
Oh yeah that's no good. It shouldn't be returning in any circumstance.